hnn_core/network.py

```diff
@@ -219,25 +227,54 @@ def __repr__(self):
               % (self.n_of_type['L2_basket'], self.n_of_type['L5_basket']))
         return '<%s | %s>' % (class_name, s)
 
-    def _create_src_list(self):
-        """ creates the immutable source list along with corresponding numbers of cells
+    def _assign_external_feeds_coords(self, n_common_feeds, p_unique_keys):
         """
-        # base source list of tuples, name and number, in this order
-        self.cellname_list = [
-            'L2_basket',
-            'L2_pyramidal',
-            'L5_basket',
-            'L5_pyramidal',
-        ]
-        self.extname_list = []
-        self.extname_list.append('common')
+        n_common_feeds : int
+            The number of common feeds.
+        p_unique_keys : list of str
+            The keys of the dictionary p_unique. Could be 'extpois',
+            'extgauss', or 'evdist_*', or 'evprox_*'
+        """
+        origin = self.pos_dict['origin']
+
+        # COMMON FEEDS
+        self.pos_dict['common'] = [origin for i in range(n_common_feeds)]
+
+        # UNIQUE FEEDS
+        for key in p_unique_keys:
+            # create the pos_dict for all the sources
+            self.pos_dict[key] = [origin for i in range(self.n_cells)]
+
+    def _add_external_feeds(self):
+        # params of common external feeds inputs in p_common
+        # Global number of external inputs ... automatic counting
+        # makes more sense
+        # p_unique represent ext inputs that are going to go to each cell
+        self.p_common, self.p_unique = create_pext(self.params,
+                                                   self.params['tstop'])
+        self.n_common_feeds = len(self.p_common)
+
+        self._assign_external_feeds_coords(self.n_common_feeds,
+                                           self.p_unique.keys())
+
+        # self._count_cells()  # sets n_cells, includes Artificial ones too!
+
+        # XXX hard-coded here until attaching feeds can be done smarter
+        self.feedname_list = []
```

Comment:
I would initialize stuff inside the init so you know at a glance what are the different attributes that could be updated in other methods

Reply:
Done. Leaving this here though as the current feed adding mechanism may call this function multiple times (ugh), so important this is reset to `[]`. Will be removed in future.

Reply:
uhmm ... I don't get you. I see only one call?

Reply:
My bad. Added some DOCs and comments but initialisation now only in __init__